app/models/goal.py

```diff
@@ -2,4 +2,25 @@
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    __tablename__ = 'goals'
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String, nullable=False)
+    tasks = db.relationship('Task', back_populates='goal', lazy=True)
+    
+    def to_dict(self):
+        return {
+                'id': self.id,
+                'title': self.title
+        }
+
+    @classmethod
+    def from_dict(cls, request_body):
+        goal = cls(
+                    title=request_body['title']
+                    )
+        return goal
+        
+    #returns all the attributes that must be passed in order to create a record. All not nullable instance variables accept id.'
+    @classmethod
+    def get_attributes(cls):
+        return ['title']
```

Comment:
I think that this method is a great idea. As you design more classes and their class/instance methods, I'd love for you to keep thinking about the trade-offs of each design decision... The trade-off here is that it's easier to use `get_attributes` to validate, but this function will need to be updated every time the `Goal` model is updated, and every model such as `Task` or any new model will need to define this method.